src/__swaps__/utils/numbers.ts

```diff
@@ -146,11 +147,29 @@ export const handleSignificantDecimalsWithThreshold = (value: BigNumberish, deci
   return lessThan(result, threshold) ? `< ${threshold}` : result;
 };
 
+export const handleSignificantDecimalsWorklet = (value: number | string, decimals: number, buffer = 3): string => {
+  'worklet';
+  let dec;
+
+  if (lessThanWorklet(value, 1)) {
+    const orderOfMagnitude = orderOfMagnitudeWorklet(value);
+    const sigDigitsWithBuffer = -orderOfMagnitude - 1 + buffer;
+    dec = Math.min(sigDigitsWithBuffer, 8);
+  } else {
+    dec = Math.min(decimals, buffer);
+  }
+  return Number(value).toLocaleString('en-US', {
+    useGrouping: true,
+    minimumFractionDigits: 2,
+    maximumFractionDigits: dec,
+  });
+};
+
 export const handleSignificantDecimals = (value: BigNumberish, decimals: number, buffer = 3, skipDecimals = false): string => {
   let dec;
   if (lessThan(new BigNumber(value).abs(), 1)) {
     dec = new BigNumber(value).toFixed()?.slice?.(2).search(/[^0]/g) + buffer;
-    dec = Math.min(decimals, 8);
+    dec = Math.min(dec, 8);
```

Comment:
fixing this as this was ported over incorrectly from the original `helpers/utilities` file: https://github.com/rainbow-me/rainbow/blob/develop/src/helpers/utilities.ts#L217

this line was ignoring the result of line 171 where it finds the index of the first significant (non-zero) digit and adds a buffer.